spec/active_record/comments_spec.rb

```diff
@@ -72,6 +72,26 @@ def capture_sql
       end
       sql.should == 'SELECT * FROM "users" WHERE "users"."id" = 1 /* xxx */'
     end
+
+    it "should be thread safe" do
+      res = []
+      ["xx", "yy"].map do |comment|
```

Comment:
i checked, test definitely fails without the patch.  not sure why you say it might be randomly green?